### PR TITLE
[Core][Proposal] Adding pSetGeometry to GeometricalObject 

### PR DESCRIPTION
**Pull request by loumalouomega**
I was trying to use the coupled geometry with the contact app. I was unable to update the geometry in the condition once this is already created. This is one option I thought, let me know if you agree

(This is #5393, but targeting master instead of core/coupling_geometry)

### CODE REVIEW

File: kratos/includes/geometrical_object.h [hunk 1]
```diff
@@ -128,6 +128,20 @@ class GeometricalObject : public IndexedObject, public Flags
     ///@name Access
     ///@{
 
+    /**
+     * @brief Sets the pointer to the geometry
+     * @param pGeometry The pointer of the geometry
+     */
+    void pSetGeometry(GeometryType::Pointer pGeometry)
+    {
+//         // TODO: This is more right?
```
Comment (loumalouomega) [inline kratos/includes/geometrical_object.h]: I don't' know if this is more right

**Review by pooyan-dadvand — COMMENTED** [review]
To be discussed in the @KratosMultiphysics/technical-committee

File: kratos/includes/geometrical_object.h (hunk 1 above)
Comment (pooyan-dadvand) [reply kratos/includes/geometrical_object.h]: The best way how it is.

File: kratos/includes/geometrical_object.h [hunk 2]
```diff
@@ -128,6 +128,20 @@ class GeometricalObject : public IndexedObject, public Flags
     ///@name Access
     ///@{
 
+    /**
+     * @brief Sets the pointer to the geometry
+     * @param pGeometry The pointer of the geometry
+     */
+    void pSetGeometry(GeometryType::Pointer pGeometry)
```
Comment (pooyan-dadvand) [inline kratos/includes/geometrical_object.h]: I would name it SetGeometry as it does not return a pointer

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/technical-committee approves the addition of the method with two conditions:

1. Making it virtual, so in the derived element one can reset the element to the new geometry. Note that elements would fail now after setting their geometry
2. Naming it as SetGeometry passing a pointer as it is.

**Comment by tteschemacher** [comment]
The changes are as suggested.

Thank you @loumalouomega!

**Review by tteschemacher — APPROVED** [review]

**Pull request merged**